pkg/sql/sem/tree/casts.go

```diff
@@ -0,0 +1,1012 @@
+	{from: types.INetFamily, to: types.StringFamily, volatility: VolatilityImmutable},
+	{from: types.JsonFamily, to: types.StringFamily, volatility: VolatilityImmutable},
+	{from: types.EnumFamily, to: types.StringFamily, volatility: VolatilityImmutable},
+
+	// Casts to CollatedStringFamily.
+	{from: types.UnknownFamily, to: types.CollatedStringFamily, volatility: VolatilityImmutable},
+	{from: types.BoolFamily, to: types.CollatedStringFamily, volatility: VolatilityImmutable},
+	{from: types.IntFamily, to: types.CollatedStringFamily, volatility: VolatilityImmutable},
+	{from: types.FloatFamily, to: types.CollatedStringFamily, volatility: VolatilityStable},
+	{from: types.DecimalFamily, to: types.CollatedStringFamily, volatility: VolatilityImmutable},
+	{from: types.StringFamily, to: types.CollatedStringFamily, volatility: VolatilityImmutable},
+	{from: types.CollatedStringFamily, to: types.CollatedStringFamily, volatility: VolatilityImmutable},
+	{from: types.BitFamily, to: types.CollatedStringFamily, volatility: VolatilityImmutable},
+	{from: types.ArrayFamily, to: types.CollatedStringFamily, volatility: VolatilityImmutable},
+	{from: types.TupleFamily, to: types.CollatedStringFamily, volatility: VolatilityImmutable},
+	{from: types.GeometryFamily, to: types.CollatedStringFamily, volatility: VolatilityImmutable},
+	{from: types.GeographyFamily, to: types.CollatedStringFamily, volatility: VolatilityImmutable},
+	{from: types.BytesFamily, to: types.CollatedStringFamily, volatility: VolatilityStable},
+	{from: types.TimestampFamily, to: types.CollatedStringFamily, volatility: VolatilityImmutable},
+	{from: types.TimestampTZFamily, to: types.CollatedStringFamily, volatility: VolatilityStable},
+	{from: types.IntervalFamily, to: types.CollatedStringFamily, volatility: VolatilityImmutable},
+	{from: types.UuidFamily, to: types.CollatedStringFamily, volatility: VolatilityImmutable},
+	{from: types.DateFamily, to: types.CollatedStringFamily, volatility: VolatilityImmutable},
+	{from: types.TimeFamily, to: types.CollatedStringFamily, volatility: VolatilityImmutable},
+	{from: types.TimeTZFamily, to: types.CollatedStringFamily, volatility: VolatilityImmutable},
+	{from: types.OidFamily, to: types.CollatedStringFamily, volatility: VolatilityImmutable},
+	{from: types.INetFamily, to: types.CollatedStringFamily, volatility: VolatilityImmutable},
+	{from: types.JsonFamily, to: types.CollatedStringFamily, volatility: VolatilityImmutable},
+	{from: types.EnumFamily, to: types.CollatedStringFamily, volatility: VolatilityImmutable},
+
+	// Casts to BytesFamily.
+	{from: types.UnknownFamily, to: types.BytesFamily, volatility: VolatilityImmutable},
+	{from: types.StringFamily, to: types.BytesFamily, volatility: VolatilityImmutable},
+	{from: types.CollatedStringFamily, to: types.BytesFamily, volatility: VolatilityImmutable},
+	{from: types.EnumFamily, to: types.BytesFamily, volatility: VolatilityImmutable},
```

Comment:
I don't think we're going to expose enum -> bytes for now (might revisit this in the future)